streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -265,14 +265,7 @@ public ByteBuffer encode() {
                 "Should never try to encode a SubscriptionInfo with version [" +
                     data.version() + "] > LATEST_SUPPORTED_VERSION [" + LATEST_SUPPORTED_VERSION + "]"
             );
-        } else {
-            final ObjectSerializationCache cache = new ObjectSerializationCache();
-            final ByteBuffer buffer = ByteBuffer.allocate(data.size(cache, (short) data.version()));
-            final ByteBufferAccessor accessor = new ByteBufferAccessor(buffer);
-            data.write(accessor, cache, (short) data.version());
-            buffer.rewind();
```

Comment:
`MessageUtil.toByteBuffer` uses `flip` whereas `rewind` is used here. Does it make any difference in this particular case?

Reply:
In this case, the state of ```flip``` is coincide with ```rewind```. This buffer is filled so the ```limit``` is always equal to ```capability``` even though ```rewind``` does not reset ```limit```.